### ISSUE TITLE
Receiving NULL when using fastapi with flutterflow

### ISSUE THREAD

**Issue by TDoggin** [tiangolo/fastapi#5473]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#this section will be for providing an API for the application

#system imports
import os
import sys
import psycopg2
import pandas as pd
import datetime as dt
import uvicorn 
from typing import Union
import fastapi 
from fastapi import FastAPI, Depends, HTTPException
from fastapi.security import OAuth2PasswordBearer
from fastapi.encoders import jsonable_encoder
from fastapi.responses import JSONResponse
import json
from starlette import status
from pydantic import BaseModel
import warnings
warnings.filterwarnings("ignore")

#the os path to the folder to the ocular-git master folder
dir_path = os.getcwd()
#the sub path to the parent folder that contains the modules to import
sub_path = '\\ocular\\database\\connection_info\\'
sys.path.append(dir_path + sub_path)
path = sys.path.append(dir_path + sub_path)

#module/package imports. example " from 'module_name' import 'function_name' " within the parent folder

from config import config

#display this programs file name and that it has been started
print(os.path.basename(__file__) , ' has started')

#logic variables
logic_name = 'API Execution Logic'

#function to perform a read query on the database
def read_postgresql_conn(name,query):
    conn = None
    try:
        params = config()
        conn = psycopg2.connect(**params)
        cur = conn.cursor()
        print("|logic| " + logic_name + " |executed query| " + query)   
        df = pd.read_sql(query, conn)   
        return df
        cur.close()
    except (Exception, psycopg2.DatabaseError) as error:
        print(error)
    finally:
        if conn is not None:
            conn.close()

#function to perform a write query on the database
def write_postgresql_conn(name,query):
    conn = None
    try:
        # read connection parameters
        params = config()
        conn = psycopg2.connect(**params)
        cur = conn.cursor()
        print("|logic| " + logic_name + " |executed query| " + query)   
        df = cur.execute(query, conn)
        conn.commit()
        cur.close()
    except (Exception, psycopg2.DatabaseError) as error:
        print(error)
    finally:
        if conn is not None:
            conn.close()

#SECURITY AUTHENTICATION LOGIC------------------------------------------------------------

# Use token based authentication
oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")

# Ensure the request is authenticated
def auth_request(token: str = Depends(oauth2_scheme)) -> bool:
    authenticated = token == os.getenv("API_KEY", "dummy-api-key")
    return authenticated

#END OF SECURITY AUTHENTICATION LOGIC------------------------------------------------------------

app = FastAPI()

#THIS IS THE SIMPLEST API THAT I CREATED FOR TROUBLESHOOTING. IT ALSO WORKS WITH POSTMAN BUT NOT FLUTTERFLOW

@app.get("/hello", status_code=status.HTTP_200_OK, dependencies=[Depends(auth_request)])
async def hello():
    return {"message": "Hello World"}
    

#THE FOLLOW APIS ARE THE ONES THAT WORK WITH POSTMAN BUT NOT FLUTTERFLOW - THESE ARE RELATED TO MY APPLICATION

@app.get("/newestcard")
async def newestcard(authenticated: bool = Depends(auth_request)):
    # Check for authentication like so
    if not authenticated:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Not authenticated")

    # query to retrieve the newest card
    retrieve_newest_card_query = 'SELECT * FROM public."API_Card_Data_View" ORDER BY "timestamp" DESC LIMIT 1'

    #call the function to query the database for the latest card to analyze
    new_card_results = read_postgresql_conn(logic_name,retrieve_newest_card_query)

    #return the results of the query as a dictionary
    return new_card_results.to_dict(orient='records')

#retrieve cards from the database based on a range of row numbers------------------------------------------------------------
@app.post("/cardrange")
async def cardrange(offset: int, limit: int, authenticated: bool = Depends(auth_request)):
    # Check for authentication like so
    if not authenticated:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Not authenticated")

    # query to retrieve the newest card. Limit is the amount of cards and offset is the starting point 0 based. App must
    # track how many cards are currently queried and where it last left off
    retrieve_card_range_query = 'SELECT * FROM public."API_Card_Data_View" ORDER BY "timestamp" DESC LIMIT ' + str(limit) + ' OFFSET ' + str(offset)

    #call the function to query the database for the latest card to analyze
    card_range_results = read_postgresql_conn(logic_name,retrieve_card_range_query)

    #return the results of the query as a dictionary
    return card_range_results.to_dict(orient='records')



#retrieve the user setpoints from the database------------------------------------------------------------

@app.get("/readsetpoints")
async def setpoints(authenticated: bool = Depends(auth_request)):
    # Check for authentication like so
    if not authenticated:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Not authenticated")

    # query to retrieve the newest card
    retrieve_setpoints_query = 'SELECT * FROM public."user_control_and_setpoints"'

    #call the function to query the database for the latest card to analyze
    setpoints_results = read_postgresql_conn(logic_name,retrieve_setpoints_query)

    #return the results with a dictionary
    return setpoints_results.to_dict('records')

#update the user setpoints in the database using post user input data with two parameters--this is used for manual shutdown and reset/ack
@app.post("/updatesetpoints")
async def setpoints(value: float, column: str, authenticated: bool = Depends(auth_request)):
    # Check for authentication like so
    if not authenticated:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Not authenticated")

    # query to update the setpoints
    update_setpoints_query = 'UPDATE public."user_control_and_setpoints" SET ' + column + ' = ' + str(value)

    #call the function to query the database for the latest card to analyze
    setpoints_results = write_postgresql_conn(logic_name,update_setpoints_query)

    #return the results with a dictionary
    return 'setpoint updated!'



#launch unvicorn to run the fastapi application on localhost with reload enabled
if __name__ == "__main__":
    uvicorn.run(app, host="localhost", port=80)
```

### Description

I am seeing an odd issue that i can't seem to track down. When i use postman I receive status code 200 and can retrieve data with api bearer token logic functioning correctly. However, when i attempt to use the flutterflow to retrieve the same data i receive a status code 200 and NULL data results. FYI, to get my localhost Fastapi outside of my network i have been using ngrok and i have verified that everything functions correctly with postman while using the ngrok external https address. I also have noticed that removing the bearer token from flutterflow still results in a status code 200. I watched the incoming traffic to uvicorn and noticed that the server is not seeing any requests when i test the flutterflow logic. i'm unsure if there is an issue on the flutterflow side or if ngrok is doing something strange with the traffic, but i'm leaning towards flutterflow since postman works perfectly with ngrok. I realize the questions here are going to be "You mention flutterflow 10 times....have you talked to them?". And yes i have a request into them as well, but i would like to get a second opinion on my logic to rule out any issues on my side while i wait to hear back from them. 

### Operating System

Windows

### Operating System Details

Python version:  3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)]
Pandas version:  1.4.3  
FastAPI version:  0.84.0
Uvicorn version:  0.18.3

Windows 10 with VSC

### FastAPI Version

FastAPI version:  0.84.0

### Python Version

Python version:  3.10.7

**Comment by TDoggin** [comment]
well to anyone stumbling across this thread with the same issue. I was able to resolve it...or it resolved itself. The only thing that has happened between now and posting this comment was i registered for the pro plan with flutterflow and paid for ngrok pro. I then used a custom domain with ngrok instead of the one provided by ngrok when using the free version. So my custom domain was "www.mydomain.api.ngrok.io" rather than the cryptic "www.ngrok.23-25-23-52-6a-3g.io". Once i did this it suddenly started working. I honestly do not know why but i have a theory that flutterflows servers only validate domains that are alpha characters and that ngroks complex temporary URL names mess up the flutterflow server. Wish i had a better explaination......i hope this helps someone out there.
